### PR TITLE
Order Expiration logging and tests

### PR DESCRIPTION
**Pull request by skamoen**
This PR adds some logging to the Expiration process to get some insight into possible errors. It also adds a test for the scheduled task responsible for the proces.

### CODE REVIEW

File: src/main/java/ch/wisv/areafiftylan/utils/TaskScheduler.java [hunk 1]
```diff
@@ -75,8 +77,10 @@ public void ExpireOrders() {
         Collection<Order> allOrdersBeforeDate = orderRepository.findAllByCreationDateTimeBefore(expireBeforeDate);
 
         List<Order> expiredOrders = allOrdersBeforeDate.stream().filter(isExpired()).collect(Collectors.toList());
-
-        expiredOrders.forEach(orderService::expireOrder);
+        if (expiredOrders.size() > 0) {
```
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/utils/TaskScheduler.java]: Is there no empty check on a list?

File: src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -74,10 +74,6 @@ public void removeAuthToken(String xAuth) {
         AuthenticationToken token =
                 authenticationTokenRepository.findByToken(xAuth).orElseThrow(XAuthTokenNotFoundException::new);
 
-        if (!token.isValid()) {
```
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationServiceImpl.java]: Why is this removed?

**Comment by skamoen** [comment]
It looks like the TaskScheduler isn't initialized on startup... Let's not merge this yet.

File: src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationServiceImpl.java (hunk 2 above)
Comment (skamoen) [reply src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationServiceImpl.java]: This just sets a boolean. The thrown exception here causes a bunch of pointless stacktraces in the log. Worst case, you set the boolean to false twice. Best case, no more stacktraces.

Comment (martijnjanssen) [reply src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationServiceImpl.java]: If the token is already not valid, why not just `return` here? Saves a write to the database and maybe sideeffects from calling `revoke()` again.

**Comment by skamoen** [comment]
Apparently there was a naming conflict with TaskScheduler in a newer Spring version. It worked in the test because of an explicit Autowired field. After the rename, it's initialized as a proper component

File: src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationServiceImpl.java (hunk 2 above)
Comment (skamoen) [reply src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationServiceImpl.java]: There could be different reasons for an invalid token. 

`isUsed() || isExpired() || isRevoked()`

An explicit revoke seems like a good idea here, even if we don't have a direct use now.
